tests/audit/report_test.py

```diff
@@ -0,0 +1,128 @@
+import os
+import random
+import string
+import tempfile
+
+import pytest
+
+from detect_secrets.audit.report import generate_report
+from detect_secrets.audit.report import SecretClassToPrint
+from detect_secrets.constants import VerifiedResult
+from detect_secrets.core import baseline
+from detect_secrets.core.secrets_collection import SecretsCollection
+from detect_secrets.plugins.basic_auth import BasicAuthDetector
+from detect_secrets.plugins.jwt import JwtTokenDetector
+from testing.factories import potential_secret_factory as original_potential_secret_factory
+
+
+CREATED_FILES = []
+
+
+@pytest.mark.parametrize(
+    'class_to_print, expected_real, expected_false',
+    [
+        (None, 2, 2),
+        (SecretClassToPrint.REAL_SECRET, 2, 0),
+        (SecretClassToPrint.FALSE_POSITIVE, 0, 3),
+    ],
+)
+def test_generate_report(class_to_print, expected_real, expected_false):
+    filename = baseline_file()
+    output = generate_report(filename, class_to_print)
+    real, false = count_results(output)
+    assert real == expected_real
+    assert false == expected_false
+    delete_all_temporal_files()
+
+
+def count_results(data):
+    real_secrets = 0
+    false_secrets = 0
+    for secret in data:
+        if SecretClassToPrint.from_class(VerifiedResult[secret['category']]) == SecretClassToPrint.REAL_SECRET:  # noqa: E501
+            real_secrets += 1
+        else:
+            false_secrets += 1
+    return real_secrets, false_secrets
+
+
+def baseline_file():
+    # Create our own SecretsCollection manually, so that we have fine-tuned control.
+    url_format = 'http://username:{}@www.example.com/auth'
+    first_secret = 'value1'
+    second_secret = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ'  # noqa: E501
+    random_secret = ''.join(random.choice(string.ascii_letters) for _ in range(8))
+    file_content = 'url = ' + url_format.format(first_secret) \
+        + '\nexample = ' + url_format.format(random_secret) \
+        + '\nlink = ' + url_format.format(first_secret)
```

Comment:
One of my favorite, not-as-well known Python module is `textwrap`. Combined with Python's f-strings, this allows you to write inline strings with much more clarity. e.g.

```suggestion
    file_content = textwrap.dedent(f"""
        url = {url_format.format(first_secret)}
        example = {url_format.format(second_secret)}
        link = {url_format.format(first_secret)} 
    """)[1:]
```